samples/pmic/native/npm1300_one_button/src/main.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/drivers/mfd/npm1300.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/sys/printk.h>
+
+#define FAST_FLASH_MS	100
+#define SLOW_FLASH_MS	500
+#define PRESS_SHORT_MS	1000
+#define PRESS_MEDIUM_MS 5000
+
+static volatile int flash_time_ms = SLOW_FLASH_MS;
+static volatile bool vbus_connected;
+
+static const struct device *pmic = DEVICE_DT_GET(DT_NODELABEL(npm1300_ek_pmic));
+static const struct device *leds = DEVICE_DT_GET(DT_NODELABEL(npm1300_ek_leds));
+static const struct device *regulators = DEVICE_DT_GET(DT_NODELABEL(npm1300_ek_regulators));
+static const struct device *ldsw = DEVICE_DT_GET(DT_NODELABEL(npm1300_ek_ldo1));
+static const struct device *charger = DEVICE_DT_GET(DT_NODELABEL(npm1300_ek_charger));
+
+static void event_callback(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	static int press_t;
+
+	if (pins & BIT(NPM1300_EVENT_SHIPHOLD_PRESS)) {
+		press_t = k_uptime_get();
+	}
+
+	if (pins & BIT(NPM1300_EVENT_SHIPHOLD_RELEASE)) {
+		press_t = k_uptime_get() - press_t;
+
+		if (press_t < PRESS_SHORT_MS) {
+			printk("Short press\n");
+			regulator_enable(ldsw);
```

Comment:
I think it's better to avoid using regulators in software side of things for this sample, to decrease complexity. Otherwise, it could be useful to pull LDO1 configuration from shield DTS to an overlay in this sample folder.

Reply:
@eisl-nordic Requested that regulator enable/disable be added, but based on other customer feedback I will update it to check enable state before calling the function.
I'll also look to update the overlay as you suggest, as the next upmerge will remove the regulator configuration from the shield overlay.